src/util/ActiveShapeUtils.tsx

```diff
@@ -63,3 +78,140 @@ export function Shape<OptionType, ExtraProps, ShapePropsType>({
 
   return shape;
 }
+
+/**
+ * This is an abstraction to handle identifying the active index from a tooltip mouse interaction
+ */
+type GraphicalItemShapeKey = 'trapezoids' | 'sectors' | 'points';
+
+type ItemDisplayName = 'Funnel' | 'Pie' | 'Scatter';
+
+type FunnelItem = {
+  x: number;
+  y: number;
+  labelViewBox: { x: number; y: number };
+  tooltipPayload: Array<{ payload: { payload: unknown } }>;
+};
+
+type PieItem = {
+  startAngle: number;
+  endAngle: number;
+  tooltipPayload: Array<{ payload: { payload: unknown } }>;
+};
+
+type ScatterItem = {
+  x: number;
+  y: number;
+  z: number;
+  payload?: unknown;
+};
+
+type ShapeData<DisplayName extends ItemDisplayName> = DisplayName extends 'Funnel'
+  ? FunnelItem
+  : DisplayName extends 'Pie'
+  ? PieItem
+  : DisplayName extends 'Scatter'
+  ? ScatterItem
+  : never;
+
+type GetActiveShapeIndexForTooltip = {
+  itemDisplayName: ItemDisplayName;
+  activeTooltipItem: ShapeData<ItemDisplayName>;
+  graphicalItem: {
+    props: Record<GraphicalItemShapeKey, unknown[]>;
+  };
+  itemData: unknown[];
+};
+
+function isFunnel(itemDisplayName: ItemDisplayName, _item: unknown): _item is FunnelItem {
+  return itemDisplayName === 'Funnel';
+}
+
+function isPie(itemDisplayName: ItemDisplayName, _item: unknown): _item is PieItem {
+  return itemDisplayName === 'Pie';
+}
+
+function isScatter(itemDisplayName: ItemDisplayName, _item: unknown): _item is ScatterItem {
+  return itemDisplayName === 'Scatter';
+}
+
+function getShapeDataKey(itemDisplayName: ItemDisplayName): GraphicalItemShapeKey {
+  let shapeKey: GraphicalItemShapeKey;
+
+  if (itemDisplayName === 'Funnel') {
+    shapeKey = 'trapezoids';
+  } else if (itemDisplayName === 'Pie') {
+    shapeKey = 'sectors';
+  } else if (itemDisplayName === 'Scatter') {
+    shapeKey = 'points';
+  }
+
+  return shapeKey;
+}
+
+/**
+ *
+ * @param {GetActiveShapeIndexForTooltip} arg an object of incoming attributes from Tooltip
+ * @returns {number}
+ *
+ * To handle possible duplicates in the data set,
+ * match both the data value of the active item to a data value on a graph item,
+ * and match the mouse coordinates of the active item to the coordinates of in a particular components shape data.
+ * This assumes equal lengths of shape objects to data items.
+ */
+export function getActiveShapeIndexForTooltip({
+  itemDisplayName,
+  activeTooltipItem,
+  graphicalItem,
+  itemData,
+}: GetActiveShapeIndexForTooltip): number {
+  const shapeKey = getShapeDataKey(itemDisplayName);
+
+  const tooltipPayload: Partial<ShapeData<typeof itemDisplayName>> = isScatter(itemDisplayName, activeTooltipItem)
+    ? activeTooltipItem.payload
+    : activeTooltipItem.tooltipPayload?.[0].payload.payload;
```

Comment:
nit: this optional chaining should probably keep going to prevent bad access 

`activeTooltipItem.tooltipPayload?.[0]?.payload?.payload`